src/pages/chartPage.js

```diff
@@ -1,12 +1,276 @@
-import React from 'react';
-import { StyleSheet, Text } from 'react-native';
+import React, {useState, useEffect} from 'react';
+import { StyleSheet, Text, Dimensions, Pressable, View} from 'react-native';
+import * as RecordStore from "dataModel/recordStore";
+import * as _ from "lodash";
+import Button from "components/button";
+import {DataBox, DataHeader} from "pages/dataPage"
+import * as Theme from "@assets/theme.json";
+import moment from "moment";
+import {
+  LineChart,
+  BarChart,
+  PieChart,
+  ProgressChart,
+  ContributionGraph,
+  StackedBarChart
+} from "react-native-chart-kit";
+import { isArrayLikeObject, isNull } from 'lodash';
+
 
 const ChartPage = () => {
-  return <Text>chart</Text>;
+  const today = new Date();
+  // I don't know how to initial records 
+  // The app will crash when touching empty ChartCanvus
+  // const [records, setRecords] = useState(
+  //   async () => await RecordStore.getRange(
+  //     moment(today).subtract(7, "days"),
+  //     moment(today).add(2, "days"),
+  //   ));
+  const [records, setRecords] = useState([])
+  const [legend, setLegend] = useState("-Day (1 week ago)")
+  
+  return( 
+    <View>
+      <Button
+        title="Fetch data a week ago"
+        icon="cloud-download-outline"
+        onPress={async () => {
+          const newRecords = await RecordStore.getRange(
+            moment(today).subtract(7, "days"),
+            moment(today).add(2, "days"),
+          );
+          setRecords(newRecords);
+          setLegend("-Day (1 week ago)");
+        }}/>
+      <Button
+      title="Fetch data a month ago"
+      icon="cloud-download-outline"
+      onPress={async () => {
+        const newRecords = await RecordStore.getRange(
+          moment(today).subtract(30, "days"),
+          moment(today).add(2, "days"),
+        );
+        setRecords(newRecords);
+        setLegend("-Day (1 month ago)");
+      }}/>
+      <Button
+      title="Fetch data 3 months ago"
+      icon="cloud-download-outline"
+      onPress={async () => {
+        const newRecords = await RecordStore.getRange(
+          moment(today).subtract(90, "days"),
+          moment(today).add(2, "days"),
```

Comment:
question: why do we still need `add(2, "days")`?

Reply:
ditto to others

Reply:
Since you have write the sample script in InputPage, and I just followed it. :D